tribe-ext-settings-import-export.php

```diff
@@ -0,0 +1,328 @@
+ *
+ *     This plugin is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+namespace Tribe\Extensions\Settings_Import_Export;
+
+use Tribe__Extension;
+
+/**
+ * Define Constants
+ */
+
+if ( ! defined( __NAMESPACE__ . '\NS' ) ) {
+	define( __NAMESPACE__ . '\NS', __NAMESPACE__ . '\\' );
+}
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if (
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( NS . 'Main' )
+) {
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Main extends Tribe__Extension {
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			// Load plugin textdomain
+			load_plugin_textdomain( 'tribe-ext-settings-import-export', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			if ( ! $this->php_version_check() ) {
+				return;
+			}
+
+			// Filters and Hooks here
+			add_action( 'admin_menu', [ $this, 'tribe_settings_menu' ], 99 );
+			add_action( 'admin_init', [ $this, 'tribe_sie_process_settings_export' ] );
+			add_action( 'admin_init', [ $this, 'tribe_sie_process_settings_import' ] );
+			add_action( 'admin_init', [ $this, 'tribe_sie_process_settings_reset' ] );
+		}
+
+		/**
+		 * Check if we have a sufficient version of PHP. Admin notice if we don't and user should see it.
+		 *
+		 * @link https://theeventscalendar.com/knowledgebase/php-version-requirement-changes/ All extensions require PHP 5.6+.
+		 *
+		 * @return bool
+		 */
+		private function php_version_check() {
+			$php_required_version = '5.6';
+
+			if ( version_compare( PHP_VERSION, $php_required_version, '<' ) ) {
+				if (
+					is_admin()
+					&& current_user_can( 'activate_plugins' )
+				) {
+					$message = '<p>';
+					$message .= sprintf( esc_html__( '%s requires PHP version %s or newer to work. Please contact your website host and inquire about updating PHP.', 'tribe-ext-settings-import-export' ), $this->get_name(), $php_required_version );
+					$message .= sprintf( ' <a href="%1$s">%1$s</a>', 'https://wordpress.org/about/requirements/' );
+					$message .= '</p>';
+					tribe_notice( 'tribe-ext-settings-import-export' . '-php-version', $message, [ 'type' => 'error' ] );
+				}
+
+				return false;
+			}
+
+			return true;
+		}
+
+		/**
+		 * Register the settings page
+		 */
+		function tribe_settings_menu() {
+			add_submenu_page(
+				'edit.php?post_type=tribe_events',
+				__( 'Settings Import / Export', 'tribe-ext-settings-import-export' ),
+				__( 'Settings Import / Export', 'tribe-ext-settings-import-export' ),
+				'manage_options',
+				'tribe_import_export', [
+					$this,
+					'tribe_sie_settings_page'
+				]
+			);
+		}
+
+		/**
+		 * Render the settings page
+		 */
+		function tribe_sie_settings_page() {
```

Comment:
Totally optional, but you could consider dropping the `tribe_sie_*` prefix from this and various other methods here. Remember that we're already within the `Tribe\Extensions\Settings_Import_Export` namespace, so the value of the prefix is far less than if we were writing an un-namespaced global function.

Reply:
...Oh, one other note that applies here and to a few other methods: our coding standards are to always be explicit about method visibility. So, in this case, you'd want:
```suggestion
		public function tribe_sie_settings_page() {
```
(As with a few other notes, not vital we fix today.)